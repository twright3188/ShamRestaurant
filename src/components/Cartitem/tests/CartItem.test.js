import React from 'react';
import renderer from 'react-test-renderer';
import { CartItem } from '../index';

it('cart item renders correctly', () => {
    const amount = [1];
    const setAmount = () => {};
    const item = {
        id: 0,
        image: 'image',
        title: 'title',
        size: 'size',
        price: 'price'
    };
    const tree = renderer.create(<CartItem image={item.image} title={item.title} id={item.id} amount={amount} size={item.size} price={item.price} setAmount={setAmount} />).toJSON();
    expect(tree).toMatchSnapshot();
});
