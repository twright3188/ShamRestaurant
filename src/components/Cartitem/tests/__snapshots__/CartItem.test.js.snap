// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart item renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "rgba(34, 38, 36, 0.15)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "image",
        }
      }
      style={
        Object {
          "borderRadius": 40,
          "height": 80,
          "marginRight": 17,
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "flexWrap": "wrap",
                "fontFamily": "Helvetica",
                "fontWeight": "400",
              },
              Object {
                "color": "#222624",
                "fontSize": 14,
                "lineHeight": 21,
              },
              Object {
                "flexWrap": "wrap",
                "marginBottom": 7,
              },
            ]
          }
        >
          title
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "flexWrap": "wrap",
                    "fontFamily": "Helvetica",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#222624",
                    "fontSize": 14,
                    "lineHeight": 21,
                  },
                  Object {
                    "color": "#78807B",
                    "marginBottom": 7,
                  },
                ]
              }
            >
              Size: 
              size
            </Text>
          </View>
          <View>
            <Text
              style={
                Array [
                  Object {
                    "flexWrap": "wrap",
                    "fontFamily": "Helvetica",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#222624",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              price
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingTop": 13,
              }
            }
          >
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(204, 50, 55, 1)",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "flexWrap": "wrap",
                    "fontFamily": "Helvetica",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#222624",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "center",
                    "width": 40,
                  },
                ]
              }
            >
              1
            </Text>
            <View
              accessible={true}
              clickable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(34, 38, 36, .4)",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
